### PR TITLE
Drill descriptions & improve MPU mechanics

### PR DESCRIPTION
**Pull request by voicey99**
Extended drill descriptions to clear up confusion. Also removed the "base" tag from the Kerbitats.

EXTENDED: Added heat mechanics onto the MPUs as well as creating a new LFO mode and increasing LF/OX/LFO/MP output by the stated 900%. Note that the 2.5m is a little slow to start, owing to a bug with the Unity float curve that meant it didn't work over a certain value (see thread p74).

### CODE REVIEW

**Comment by Wyzard256** [comment]
Testing out this change, I loaded a craft whose MPU had been configured for Fertilizer, and it came up configured for Polymers instead.  Since the new LFO converter was added at the beginning of the module list, it changed the indexes of all the others, and the MODULE entries in a `.craft` or `.sfs` file now match up with a different module than they used to.  The Fertilizer(G) converter used to be the seventh instance of `ModuleResourceConverter_USI`, so my `.craft` file has the seventh one marked as enabled, but now the seventh converter is Polymers, and Fertilizer(G) is the eighth.

In my case, this happened in the VAB and was easy enough to fix, but it could be an unpleasant surprise for people who have MPUs out in space where they need SpecializedParts and an engineer to reconfigure.

**Comment by voicey99** [comment]
Would moving the Minerals->Fertiliser line to the end (but still before the efficiency modes) mitigate this change by shifting the rest back up one again? If no, problems like this are why changes like this are good to implement before the parts in question come into mainstream use.

**Comment by Wyzard256** [comment]
Fertilizer(M) is already the last converter.  Did you mean to move something from the beginning?

One option that would minimize the impact is:
* Move the MonoPropellant converter to the end, just after Fertilizer(M).
* Move the new LFO converter to be after the LiquidFuel and Oxidizer converters, so it's in the spot where MonoPropellant used to be.

That way, MPUs configured for monopropellant would change to LFO, but everything else should be unaffected.  (I haven't tested this, though.)

On the other hand, the addition of heat generation may also "break" existing vessels (if they don't have enough radiators), so maybe this just comes down to the fact that MKS is still in development and backward compatibility is not guaranteed.  I personally wouldn't mind just hand-editing my save file to change the active converter back to what it should be.

**Comment by voicey99** [comment]
I actually meant moving the chems converter (was on mobile, had to guess whether it was fert or chems) to the end, and doing that would keep the fuel "block" together so all the fuel options are next to one another. while preserving the majority of the other options' positions in the list.
As for the radiators, something I've seen with a lot of bases is that most have a Ranger TCS to cool the drills and other modules, and a TCS should be able to handle the MPUs just fine.

**Comment by Wyzard256** [comment]
In that case, yeah, moving the chemicals converter to the end would resolve the problem for the converters after it.  It's similar to what I suggested with the monopropellant converter, except that I wasn't thinking about wanting to keep monopropellant together with the other fuel converters.

On a separate note: I just noticed that you said in your comment up top that the 2.5m MPU is slow to start due to an issue with the float curve.  I noticed that in my testing on the launchpad, but I also found that the 2.5m MPU doesn't work at all on the Mun — only heats up a little bit, and thermal efficiency stays at 0%.  Depending on kolonization bonus, it may start out slightly above 0% but then it goes *down* as heat goes up.  I don't know whether it's a problem with the float curve or some sort of code bug in `ModuleResourceConverter_USI`, but you may want to take a look at #1199.

**Comment by voicey99** [comment]
Not sure what can be done about that one,  as raising the heat key breaks it completely (the heat generation crashes above 500K so the heating slows to a stop at precisely 564.24K every time when the 500K key is set to 25000) and lowering it slows down startup times to beyond the limit of sanity. Unity isn't my field but I'm going to guess it's the culprit here. Strange that its just the 2.5m though, the 1.25 and 3.75m versions have no issues that I found during pre-PR testing (but there's no bee-free guarantee).

**Comment by Wyzard256** [comment]
I see what's causing the heating problem.  The `ThermalEfficiency` curve is very close to zero at the MPU's starting temperature (247K in my test save on the Mun):

curve-bad

With zero efficiency, the converter runs with zero load, so it doesn't produce any heat, so it never gets itself to higher temperatures where its efficiency is above zero.  Testing on the launchpad works because Kerbin is warmer than the Mun, so the starting temperature is high enough to get around 0.2% efficiency — not much, but enough to produce a little heat and gradually raise the temperature.  (This is why the converter is so slow to ramp up thermally.)

Looking at the configs for some other parts with resource converters (like the stock ISRU), I noticed that they all have an additional "0 0" at the end of each `key` line in the `ThermalEfficiency` curve.  Those are tangent-in and tangent-out parameters, as explained [here](http://forum.kerbalspaceprogram.com/index.php?/topic/84201-info-ksp-floatcurves-and-you-the-magic-of-tangents/), and if you add them to the MPU's curve, it turns into this:

curve-good

Now the efficiency at the starting temperature is higher, so the converter is able to heat itself more quickly, and it'll work anywhere that the temperature isn't absolute zero.

Bottom line: add "0 0" to the end of each `key` line in all the `ThermalEfficiency` blocks.  (Different values can work too, but zero slope at the key points seems to be the established convention for resource converters.)

**Comment by Wyzard256** [comment]
One other thing: I notice that the 1.25m MPU's converters have `AutoShutdown = true` added, but the other two sizes don't.  Just thought I'd mention it in case the discrepancy wasn't intentional.

**Comment by voicey99** [comment]
I'll test this one out tomorrow and if it works I'll re-PR it in. Also, what program are you using to visualise the key curves?

**Comment by Wyzard256** [comment]
It's a Unity package: [r4m0n's FloatCurve Editor](http://forum.kerbalspaceprogram.com/index.php?/topic/75187-unity-floatcurve-editor-10/).

(If you copy the `key` lines from a `.cfg` into the editor, note that you have to remove the tabs at the beginning of the lines or it won't parse them.)

**Comment by voicey99** [comment]
I'll add AutoShutdown tags and bump up the 2.5m's efficiency at 500K to 0.2, should be good (gives a much smoother curve than adding 0 0, all the other MPU curves look fine-the reason the 1.25m has a 500K key of 1.25 is because I went with the stock miniISRU in that it is effective over a pretty wide range of temperatures). Thanks for the heads-up (I'm still very new to this). Given the crazy Geology bonuses, I might have to raise the max cooling limit too-did you spot any other issues in the heat mechanics?

**Comment by Wyzard256** [comment]
I sent a bunch of automated refineries to the Mun yesterday using the "0 0" version of the efficiency curve, and they've been working fine; I haven't noticed any other problems.  My geology bonus there is only around 111%, though, so I don't know if large bonuses would be problematic.  I'll try out your changes from #1206 later and let you know if anything goes wrong.

If you do raise the cooling limit, BTW, be careful of #1198.  (it's currently caused by USI drills, but could probably be cause by the MPUs too.)
